lib/internal/modules/esm/resolve.js

```diff
@@ -376,6 +377,13 @@ function finalizeResolution(resolved, base, preserveSymlinks) {
 
   let path = fileURLToPath(resolved);
   if (getOptionValue('--experimental-specifier-resolution') === 'node') {
+    if (!experimentalSpecifierResolutionWarned) {
+      process.emitWarning(
+        'The Node.js specifier resolution flag is experimental. It could change or be removed at any time.',
+        'ExperimentalWarning');
+      experimentalSpecifierResolutionWarned = true;
+    }
```

Comment:
Should we maybe hoist this into ESMLoader itself, perhaps in the constructor?

Reply:
Sure, is there a spot in particular that you think it should go?

Reply:
There currently is no constructor in ESMLoader, so you could add one and we could put all these experimental warning sorts of checks in there (not necessarily moving them all there in this PR, but maybe a good idea to do that together?).

Reply:
Okay! I shall do this weekend.